src/riffvideo.cpp

```diff
@@ -0,0 +1,1287 @@
+    Exiv2::XmpTextValue tv(Exiv2::toString(Exiv2::getLong(buf.data(), littleEndian)));
+    xmpData_.add(Exiv2::XmpKey("Xmp.xmp.Thumbnails/xmpGImg:width"), &tv);
+
+    io_->read(buf.data(), 2);
+    tv.read(Exiv2::toString(Exiv2::getLong(buf.data(), littleEndian)));
+    xmpData_.add(Exiv2::XmpKey("Xmp.xmp.Thumbnails/xmpGImg:height"), &tv);
+
+    io_->read(buf.data(), 4);
+
+    /* TODO - Storing the image Thumbnail in Base64 Format
+
+
+    uint64_t length = Exiv2::getLong(buf.data(), littleEndian);
+    io_->read(buf.data(), length);
+
+    char *rawStr = Exiv2::toString(buf.data());
+    char *encodedStr;
+
+    SXMPUtils::EncodeToBase64(rawStr, encodedStr);
+
+    tv.read(Exiv2::toString(encodedStr));
+    xmpData_.add(Exiv2::XmpKey("Xmp.xmp.Thumbnails/xmpGImg:image"), &tv);
+*/
+  } else {
+    io_->seek(cur_pos, BasicIo::beg);
+    io_->read(buf.data(), size);
+    xmpData_["Xmp.video.Junk"] = buf.data();
+  }
+
+  io_->seek(cur_pos + size, BasicIo::beg);
+}  // RiffVideo::junkHandler
+
+void RiffVideo::aviHeaderTagsHandler(size_t size) {
+  const size_t bufMinSize = 4;
+  DataBuf buf(bufMinSize + 1);
+  buf.data()[4] = '\0';
+  size_t width = 0, height = 0, frame_count = 0;
+  double frame_rate = 1;
+
+  uint64_t cur_pos = io_->tell();
+
+  for (int i = 0; i <= 9; i++) {
+    std::memset(buf.data(), 0x0, buf.size());
+    io_->read(buf.data(), bufMinSize);
+
+    switch (i) {
+      case frameRate:
+        xmpData_["Xmp.video.MicroSecPerFrame"] = Exiv2::getULong(buf.data(), littleEndian);
+        frame_rate = (double)1000000 / (double)Exiv2::getULong(buf.data(), littleEndian);
+        break;
+      case (maxDataRate):
+        xmpData_["Xmp.video.MaxDataRate"] = (double)Exiv2::getULong(buf.data(), littleEndian) / (double)1024;
+        break;
+      case frameCount:
+        xmpData_["Xmp.video.FrameCount"] = Exiv2::getULong(buf.data(), littleEndian);
+        frame_count = Exiv2::getULong(buf.data(), littleEndian);
+        break;
+      case streamCount:
+        xmpData_["Xmp.video.StreamCount"] = Exiv2::getULong(buf.data(), littleEndian);
+        break;
+      case imageWidth_h:
+        width = Exiv2::getULong(buf.data(), littleEndian);
+        xmpData_["Xmp.video.Width"] = width;
+        break;
+      case imageHeight_h:
+        height = Exiv2::getULong(buf.data(), littleEndian);
+        xmpData_["Xmp.video.Height"] = height;
+        break;
+    }
+  }
+
+  fillAspectRatio(width, height);
+  fillDuration(frame_rate, frame_count);
+
+  io_->seek(cur_pos + size, BasicIo::beg);
+}  // RiffVideo::aviHeaderTagsHandler
+
+void RiffVideo::streamHandler(size_t size) {
+  const size_t bufMinSize = 4;
+  DataBuf buf(bufMinSize + 1);
+  buf.data()[4] = '\0';
+  size_t divisor = 1;
+  uint64_t cur_pos = io_->tell();
+
+  io_->read(buf.data(), bufMinSize);
+  if (equalsRiffTag(buf, "VIDS"))
+    streamType_ = Video;
+  else if (equalsRiffTag(buf, "AUDS"))
+    streamType_ = Audio;
+
+  for (int i = 1; i <= 25; i++) {
```

Comment:
You are right, I shouldn't iterate till 24, but I need to iterate sequentially till the last value in the enum, even It the switch does not have all values.
because  in io_->read(buf.data(), RIFF_TAG_SIZE);   the position is advanced by the number of bytes read.